cmd/create/machinepool/nodepool.go

```diff
@@ -493,3 +499,17 @@ func createAwsNodePoolBuilder(instanceType string, securityGroupIds []string) *c
 
 	return awsNpBuilder
 }
+
+func isCompatibleVersionForAdditionalSecurityGroupsIds(version string) (bool, error) {
```

Comment:
@oriAdler How about we turn this into something generic that can be reused throughout the CLI and then easily swapped out if we move the implementation to the backend?

I was thinking something like:

```
package features

import (
	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
	"github.com/openshift/rosa/pkg/helper/versions"
)

type OCMFeature string

const (
	AdditionalDay2SecurityGroupsHcpFeature OCMFeature = "AdditionSecurityGroupsDay2HCP"
)

var ocmFeatureVersions = map[OCMFeature]string{
	AdditionalDay2SecurityGroupsHcpFeature: "4.15.0-0.a",
}

type FeatureService interface {
	IsFeatureSupported(feature OCMFeature, cluster *cmv1.Cluster) (bool, error)
}

type FeatureServiceImpl struct {
	featureVersionMap map[OCMFeature]string
}

func NewFeatureService() FeatureService {
	return &FeatureServiceImpl{featureVersionMap: ocmFeatureVersions}
}

func (f *FeatureServiceImpl) IsFeatureSupported(feature OCMFeature, cluster *cmv1.Cluster) (bool, error) {
	supportedVersion := f.featureVersionMap[feature]
	if supportedVersion != "" {
		return versions.IsGreaterThanOrEqual(cluster.Version().RawID(), supportedVersion)
	}

	return true, nil
}
```

And then, your caller looks like:

```
featureService := NewFeatureService()
isSupported, err := featureService.IsFeatureSupported(AdditionalDay2SecurityGroupsHcpFeature, cluster)
```

WDYT?

Reply:
Thanks a lot for the feedback.
I like the approach of a generic function to check the feature version.

With that being said, I'm not sure we use the `Service` pattern in the CLI, usually more common in the backend.
Created a new `features` package, and added a general function to the package.

Moreover, added a note to `versions.go` to add new minimal versions to the new package, we can also refactor the existing checks as part of our tech debt.

Please let me know if the solution seems valid to you, and I can create a ticket to move existing minimal version checks to use the `features` package.

Reply:
@oriAdler First implementation looks good to me 👍 

I've left a comment about us updating `CONTRIBUTING.md` and also communicating to the team about this change. We want to ensure that everyone knows this exists and this is how we want to enforce version gating of features in the CLI at this time, until we can fully remove it and rely on the backend.

+1 to creating another issue to move existing feature gates to this approach. Sounds like `good-first-issue` to me 👍 

cheers,

Rob